eth/filters/api.go

```diff
@@ -239,13 +239,13 @@ func (args *NewFilterArgs) UnmarshalJSON(data []byte) error {
 		return err
 	}
 
-	if raw.From == nil {
+	if raw.From == nil || raw.From.Int64() < 0 {
```

Comment:
This is specifically true for the ToBlock field. I'm not sure whether From needs it or not, but we definitely need to convert the API "special block numbers" to "filter numbers" properly.